index.js

```diff
@@ -0,0 +1,43 @@
+const express = require("express");
+const knex = require("knex");
+const bcrypt = require('bcryptjs');
+
+const knexConfig = require("./knexfile.js");
+
+const db = knex(knexConfig.development);
+
+const server = express();
+server.use(express.json());
+
+// get all users
+server.get('/api/users', (req, res) => {
+    db('users').select('id', 'username', 'password').then(users => {
+        res.json(users);
+    }).catch(err => res.send(err));
+});
+
+// register user
+server.post('/api/register', (req, res) => {
+    const creds = req.body;
+    const hash = bcrypt.hashSync(creds.password, 14);
+    creds.password = hash;
+
+    db('users').insert(creds).then(ids => {
+        res.status(201).json(ids);
+    }).catch(err => json(err));
```

Comment:
you need to do `res.status(400).json(err)` or something like that, make sure you're testing your routes in Postman